src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultUIContextManager.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Threading;
+
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Export(typeof(IUIContextManager))]
+    public class DefaultUIContextManager : IUIContextManager
+    {
+        private readonly IServiceProvider _serviceProvider;
+        private readonly JoinableTaskFactory _joinableTaskFactory;
+
+        [ImportingConstructor]
+        public DefaultUIContextManager(SVsServiceProvider serviceProvider, JoinableTaskContext joinableTaskContext)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            if (joinableTaskContext is null)
+            {
+                throw new ArgumentNullException(nameof(joinableTaskContext));
+            }
+
+            _serviceProvider = serviceProvider;
+            _joinableTaskFactory = joinableTaskContext.Factory;
+        }
+
+        public async Task SetUIContextAsync(Guid uiContextGuid, bool isActive, CancellationToken cancellationToken)
+        {
+            await _joinableTaskFactory.SwitchToMainThreadAsync();
+
+            var monitorSelection = _serviceProvider.GetService(typeof(SVsShellMonitorSelection)) as IVsMonitorSelection;
+            Assumes.Present(monitorSelection);
+            var cookieResult = monitorSelection.GetCmdUIContextCookie(uiContextGuid, out var cookie);
+            _ = ErrorHandler.ThrowOnFailure(cookieResult);
```

Comment:
lol does the `ErrorHandler.ThrowOnFailure` return a code indicating if it was able to successfully throw on failure?

Reply:
It returns the HResult, but that could only be 0 unless it threw. Honestly I only did this because VS started complaining about IDE0058, which really isn't that useful anyway. I've opted to instead remove theses `_ =` where they don't reference a Task and silence IDE0058 in the `.editorconfig`.